pkg/etcd/etcd.go

```diff
@@ -329,6 +329,15 @@ func (e *ETCD) join(ctx context.Context, clientAccessInfo *clientaccess.Info) er
 	}
 
 	for _, member := range members.Members {
+		memberNodeName := strings.Split(member.Name, "-")[0]
```

Comment:
How will this code behave with hostnames that contain hyphens?

Reply:
+1
> Valid characters for hostnames are ASCII(7) letters from a to z, the digits from 0 to 9, and the hyphen (-). A hostname may not start with a hyphen.